### PR TITLE
[DLSP21] 02-3

### PR DESCRIPTION
**Pull request by rajashekarv95**
Hi Alf, Can you please review this PR and let me know if I have to make any changes?

### CODE REVIEW

File: docs/en/week02/02-3.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+## Resources
+
+Please follow Alfredo Canziani [on Twitter @alfcnz](https://twitter.com/alfcnz). Videos and textbooks with relevant details on linear algebra and singular value decomposition (SVD) can be found by searching Alfredo's Twitter, for example type `linear algebra (from:alfcnz)` in the search box.
+
+
+## [Neural Nets: Rotation and Squashing](https://youtu.be/0TdAmZUMj2k)
+A traditional neural network is an alternating collection of two blocks - the linear blocks and the non-linear blocks. Given below is a block diagram of a traditional neural network. 
+
+<center><img src="{{site.baseurl}}/images/week02/02-3/figure1.png" width="1000px"/></center>
```
Comment (Atcold) [inline docs/en/week02/02-3.md:18]: Need caption.

File: docs/en/week02/02-3.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+The linear blocks(Rotations, for simplicity) are given by:
+
+$$ 
+s_{k+1} = w_kz_k
```
Comment (Atcold) [inline docs/en/week02/02-3.md:23]: Please, use `\vect{}` for all vectors.

File: docs/en/week02/02-3.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+Please follow Alfredo Canziani [on Twitter @alfcnz](https://twitter.com/alfcnz). Videos and textbooks with relevant details on linear algebra and singular value decomposition (SVD) can be found by searching Alfredo's Twitter, for example type `linear algebra (from:alfcnz)` in the search box.
+
+
+## [Neural Nets: Rotation and Squashing](https://youtu.be/0TdAmZUMj2k)
+A traditional neural network is an alternating collection of two blocks - the linear blocks and the non-linear blocks. Given below is a block diagram of a traditional neural network. 
+
+<center><img src="{{site.baseurl}}/images/week02/02-3/figure1.png" width="1000px"/></center>
+
+The linear blocks(Rotations, for simplicity) are given by:
```
Comment (Atcold) [inline docs/en/week02/02-3.md:20]: Add a space before `(`, here and everywhere else.

File: docs/en/week02/02-3.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
+---
+lang-ref: ch.01-3
+title: Problem Motivation, Linear Algebra, and Visualization
+lecturer: Alfredo Canziani
+authors: Rajashekar Vasantha, Derek Yen, Tony Xu, Ben Stadnick, Prasanthi Gurumurthy
```
Comment (Atcold) [inline docs/en/week02/02-3.md:5]:
```suggestion
authors: Rajashekar Vasantha
```

**Review by Atcold — CHANGES_REQUESTED** [review]
Good, a few edits.
Summary is missing in`en/week02/02.md`.
Description of the warping net and final vectors is missing.

File: docs/en/week02/02-3.md (hunk 1 above)
Comment (rajashekarv95) [reply docs/en/week02/02-3.md]: Added caption

File: docs/en/week02/02-3.md (hunk 2 above)
Comment (rajashekarv95) [reply docs/en/week02/02-3.md]: Added \vect{}

File: docs/en/week02/02-3.md (hunk 3 above)
Comment (rajashekarv95) [reply docs/en/week02/02-3.md]: Added spaces everywhere

**Comment by rajashekarv95** [comment]
Hi Alf, I have made the changes suggested by you. While making the changes, I realized that W is not capitalized in the block diagram. Should I make a new one to ensure consistency?

**Comment by Atcold** [comment]
> Hi Alf, I have made the changes suggested by you. While making the changes, I realized that W is not capitalized in the block diagram. Should I make a new one to ensure consistency?

Yes plz.

**Comment by rajashekarv95** [comment]
Hi Alf, I have added the new diagram. Can you please review it?

**Review by Atcold — CHANGES_REQUESTED** [review]
It's good but they should be bold 😬
Use `\matr{}` for matrices and `\vect{}` for vectors.

Then we're good here.

**Comment by rajashekarv95** [comment]
I have made the above changes, but wasn't able to use \vect and \matr in the diagram - but have added everywhere in the markdown. Please let me know if there is any way to do this on diagrams.net, I will make those changes as well.

**Comment by Atcold** [comment]
Yeah, you need to use `\boldsymbol{}` there. Sorry for not been clear.

File: docs/en/week02/02-3.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+$$ 
+\vect{s}_{k+1} = \matr{W}_k\vect{z}_k 
+$$
+
+And the non-linear blocks (Squashing functions for intuitive understanding) are given by:
+
+$$ \vect{z}_k = h(\vect{s}_k) $$
+
+In the above diagram and equations, $$\vect{x} \in \mathbb{R}^n$$ represents the input vector. $$\matr{W}_k \in \mathbb{R}^{n_{k} \times n_{k-1}}$$ represents the matrix of an affine transformation corresponding to the $$k^{\text{th}}$$ block and is described below in further detail. The function $h$ is called the activation function and this function forms the non-linear block of the neural network. Sigmoid, ReLu and tanh are some of the common activation functions and we will look at them in the later parts of this section. After alternate applications of linear and non-linear blocks, the above network produces an output vector $$\vect{s_k} \in \mathbb{R}^{n_{k-1}}$$.
```
Comment (Atcold) [inline docs/en/week02/02-3.md:34]:
```suggestion
In the above diagram and equations, $$\vect{x} \in \mathbb{R}^n$$ represents the input vector. $$\matr{W}_k \in \mathbb{R}^{n_{k} \times n_{k-1}}$$ represents the matrix of an affine transformation corresponding to the $$k^{\text{th}}$$ block and is described below in further detail. The function $h$ is called the activation function and this function forms the non-linear block of the neural network. Sigmoid, ReLu and tanh are some of the common activation functions and we will look at them in the later parts of this section. After alternate applications of linear and non-linear blocks, the above network produces an output vector $$\vect{s}_k \in \mathbb{R}^{n_{k-1}}$$.
```

**Review by Atcold — CHANGES_REQUESTED** [review]

**Comment by rajashekarv95** [comment]
Thank you for the suggestions, I have made the changes and pushed them.

**Comment by Atcold** [comment]
Thanks!

**Pull request merged**